ush/rocoto/setup_expt_fcstonly.py

```diff
@@ -207,6 +207,23 @@ def edit_baseconfig():
       chgrp_rstprod = 'YES'
       chgrp_cmd = 'chgrp rstprod'
       hpssarch = 'YES'
+    elif machine == 'JET':
+      base_git = '/lfs4/HFIP/hfv3gfs/glopara/git'
+      base_svn = '/dev/null/global/save/glopara/svn/'
+      dmpdir = '/lfs4/HFIP/hfv3gfs/glopara/dump'
+      nwprod = '/lfs4/HFIP/hfv3gfs/glopara/nwpara'
+      comroot = '/lfs4/HFIP/hfv3gfs/glopara/com'
+      homedir = '/lfs4/HFIP/hfv3gfs/$USER'
+      stmp = '/lfs4/HFIP/hfv3gfs/${USER}/stmp'
+      ptmp = '/lfs4/HFIP/hfv3gfs/${USER}/ptmp'
+      noscrub = '$HOMEDIR'
+      account = 'hfv3gfs'
+      queue = 'batch'
+      queue_service = 'service'
+      partition_batch = '$PARTITION_BATCH'
```

Comment:
`$PARTITION_BATCH` needs to be substituted with a value so a default needs to be provided.
```suggestion
      partition_batch = "${PARTITION_BATCH:-'xjet'}"
```

Reply:
Why default everyone to use xjet on JET? I believe if a default is need, the python script should be modified to require config.base to have PARTITION set on JET as part of exception handling. May I keep this solution without change at this time?

Reply:
That is not true.
This line will add:
```
export PARTITION_BATCH=${PARTITION_BATCH:-'xjet'}
```
to `config.base` instead of:
```
export PARTITION_BATCH=$PARTITION_BATCH
```

If you don't provide a value, what is the value the script will pick?

For the user to use a specific partition, he user **has** to modify this line in `config.base` to provide a value or set it in the environment.

Reply:
PARTITION_BATCH in config.base REQUIRES a value so when PARTITION_BATCH goes through config.resources (as sourced by the setup_workflow*py scripts) it has to have a value for the Jet if-block in there. As it is right now the setup_workflow_fcstonly.py script fails:
```
-bash-4.2$ ./setup_workflow_fcstonly.py --expdir /lfs4/HFIP/hfv3gfs/Kate.Friedman/expdir/pr394ff
sourcing config.getic
sourcing config.init
sourcing config.waveinit
(standard_in) 1: syntax error
sourcing config.waveprep
(standard_in) 1: syntax error
sourcing config.fcst
(standard_in) 1: syntax error
(standard_in) 2: syntax error
(standard_in) 2: syntax error
sourcing config.post
sourcing config.wavepostsbs
(standard_in) 1: syntax error
sourcing config.wavepostbndpnt
(standard_in) 1: syntax error
sourcing config.wavepostbndpntbll
(standard_in) 1: syntax error
sourcing config.wavepostpnt
(standard_in) 1: syntax error
sourcing config.wavegempak
(standard_in) 1: syntax error
sourcing config.waveawipsbulls
(standard_in) 1: syntax error
sourcing config.waveawipsgridded
(standard_in) 1: syntax error
sourcing config.wafs
sourcing config.wafsgrib2
sourcing config.wafsblending
sourcing config.wafsgcip
sourcing config.wafsgrib20p25
sourcing config.wafsblending0p25
sourcing config.postsnd
sourcing config.gempak
sourcing config.awips
sourcing config.vrfy
sourcing config.metp
sourcing config.arch
Traceback (most recent call last):
  File "./setup_workflow_fcstonly.py", line 774, in <module>
    main()
  File "./setup_workflow_fcstonly.py", line 54, in main
    create_xml(dict_configs)
  File "./setup_workflow_fcstonly.py", line 744, in create_xml
    resources = get_resources(dict_configs, cdump=base['CDUMP'])
  File "./setup_workflow_fcstonly.py", line 171, in get_resources
    wtimestr, resstr, queuestr, memstr, natstr = wfu.get_resources(machine, cfg, task, reservation, cdump=cdump)
  File "/mnt/lfs4/HFIP/hfv3gfs/Kate.Friedman/git/feature-jet-fcst-only/ush/rocoto/workflow_utils.py", line 309, in get_resources
    nodes = np.int(np.ceil(np.float(tasks) / np.float(ppn)))
ValueError: could not convert string to float:
```

We need a default in PARTITION_BATCH in config.base....is there a desired default partition? xjet? kjet? other? Users can always change the partition in config.base before running setup_workflow*py to whichever partition they want to use other than whichever default we set.

Did you run the setup scripts from a cleanly cloned/built copy of this branch?

Reply:
> As it is right now the setup_workflow_fcstonly.py script fails:
@KateFriedman-NOAA 
Please remove posted message for this session. The testing of setup_workflow_fcstonly.py should use python3.

Reply:
> That is not true.
> This line will add:
> 
> ```
> export PARTITION_BATCH=${PARTITION_BATCH:-'xjet'}
> ```
> 
> to `config.base` instead of:
> 
> ```
> export PARTITION_BATCH=$PARTITION_BATCH
> ```
> 
> If you don't provide a value, what is the value the script will pick?
> 
> For the user to use a specific partition, he user **has** to modify this line in `config.base` to provide a value or set it in the environment.

I am making change base on your request and testing it now.

Reply:
I also get a syntax error with `export PARTITION_BATCH="${PARTITION_BATCH:-'xjet'}"` in config.base when setup_workflow_fcstonly.py sources it.

I spoke with @GeorgeVandenberghe-NOAA yesterday and he recommends we stick with just `xjet` for this initial pass at supporting Jet. The login nodes build for `xjet` and `kjet` by default right now so users who wish to use the other partitions will need to build for other partitions themselves for now. I ran my tests yesterday with `xjet` and the init/fcst/post jobs worked (getic and arch jobs used service queue). Users will want to turn off vrfy and metp jobs with DO_VRFY=NO and DO_METP=NO.

Please change the line in question in setup_expt_fcstonly.py to be:
```
partition_batch = 'xjet'
```

Thanks!

Reply:
> ush/rocoto/rocoto_viewer.py

File rocoto_viewer.py was not supported in PR 370 that convert python2 to python3.
rocoto_viewer.py should get the same translation done before working on JET version.